edgeml_pytorch/trainer/bonsaiTrainer.py

```diff
@@ -126,13 +126,13 @@ def runHardThrsd(self):
             __thrsdT).to(self.device)
 
         self.__thrsdW = torch.FloatTensor(
-            __thrsdW.clone()).to(self.device)
+            __thrsdW.clone().detach()).to(self.device)
```

Comment:
@pushkalkatara it needs to be `.detach().clone()`. Can you please change it.

Reply:
@adityakusupati made the change with recent commit.